src/components/CloudHeader/CloudHeader.js

```diff
@@ -0,0 +1,122 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import ReactDOM from 'react-dom';
+import classNames from 'classnames';
+import CloudHeaderWrapper from './CloudHeaderWrapper';
+import CloudHeaderMenu from './CloudHeaderMenu';
+import CloudHeaderLogo from './CloudHeaderLogo';
+import CloudHeaderList from './CloudHeaderList';
+import CloudHeaderListItem from './CloudHeaderListItem';
+
+export default class CloudHeaderV2 extends React.Component {
+  state = {
+    isMenuOpen: false,
+  };
+
+  handleMenuClick = () => {
+    this.setState({
+      isMenuOpen: !this.state.isMenuOpen,
+    });
+  };
+
+  componentDidMount() {
+    this.portalNode = document.createElement('div');
+    document.body.appendChild(this.portalNode);
+  }
+
+  componentWillUnmount() {
+    this.portalNode.parentNode.removeChild(this.portalNode);
+  }
+
+  render() {
+    const {
+      className,
+      companyName,
+      productName,
+      logoHref,
+      renderMenu,
+      renderLogo,
+      renderSearch,
+      renderNotification,
+      renderApplications,
+      renderUser,
+      links,
+      ...other
+    } = this.props;
+    const { isMenuOpen } = this.state;
+    const cloudHeaderClasses = classNames('bx--cloud-header', className);
+
+    return (
+      <nav key="nav" className={cloudHeaderClasses} {...other}>
+        <CloudHeaderWrapper>
+          {renderMenu && <CloudHeaderMenu onClick={this.handleMenuClick} />}
+          <CloudHeaderLogo
+            className={!renderMenu && 'bx--cloud-header-brand--no-menu'}
+            companyName={companyName}
+            productName={productName}
+            href={logoHref}>
+            {renderLogo && renderLogo()}
+          </CloudHeaderLogo>
+          <CloudHeaderList>
+            {links &&
+              links.map(link => {
+                return (
+                  <CloudHeaderListItem href={link.href}>
+                    {link.linkText}
+                  </CloudHeaderListItem>
+                );
+              })}
+          </CloudHeaderList>
+        </CloudHeaderWrapper>
+        <CloudHeaderWrapper>
+          <CloudHeaderList>
+            {renderSearch && (
+              <CloudHeaderListItem onClick={renderSearch} isIcon>
+                <svg viewBox="0 0 16 16" fillRule="evenodd">
+                  <title>search</title>
+                  <path d="M6 2c2.2 0 4 1.8 4 4s-1.8 4-4 4-4-1.8-4-4 1.8-4 4-4zm0-2C2.7 0 0 2.7 0 6s2.7 6 6 6 6-2.7 6-6-2.7-6-6-6zm10 13.8L13.8 16l-3.6-3.6 2.2-2.2z" />
+                  <path d="M16 13.8L13.8 16l-3.6-3.6 2.2-2.2z" />
+                </svg>
+              </CloudHeaderListItem>
+            )}
+            {renderNotification && (
+              <CloudHeaderListItem onClick={renderNotification} isIcon>
```

Comment:
Should we have header state managing notifications/applications/account stuff as well?

Reply:
Yeah, might as well add that in if we have that for `Menu` 👍